app/Http/Controllers/Shopify/AppController.php

```diff
@@ -225,158 +225,162 @@ public function printLabels(Request $request) {
                 continue;
             }
 
-            $done_shipment = ShopifyShipment::where('shop_id', $shop->id)
-                    ->where('order_id', $order['legacyResourceId'])
-                    ->where('test_mode', $shop->test_mode)
-                    ->where('return', $is_return)
-                    ->first();
+            DB::transaction(function () {
```

Comment:
Should have also use ($shop, etc variables that are needed inside this function call...